citesphere/src/main/java/edu/asu/diging/citesphere/core/authority/impl/ConceptpowerImporter.java

```diff
@@ -0,0 +1,172 @@
+        }
+
+        return false;
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see edu.asu.diging.citesphere.authority.impl.AuthorityImporter#
+     * retrieveAuthorityData(java.lang.String)
+     */
+    @Override
+    public IImportedAuthority retrieveAuthorityData(String searchString)
+            throws URISyntaxException, AuthorityServiceConnectionException {
+        return null;
+    }
+
+    @Override
+    public String getId() {
+        return ID;
+    }
+
+    @Override
+    public List<IAuthorityEntry> retrieveAuthoritiesData(String searchString, int page, int pageSize)
+            throws URISyntaxException, AuthorityServiceConnectionException {
+
+        searchString = searchString.replace(conceptpowerSearchKeyword, "");
+        HttpComponentsClientHttpRequestFactory factory = new HttpComponentsClientHttpRequestFactory();
+        HttpClient httpClient = HttpClientBuilder.create().setRedirectStrategy(new LaxRedirectStrategy()).build();
+        factory.setHttpClient(httpClient);
+        restTemplate.setRequestFactory(factory);
+
+        RequestEntity<Void> request;
+        try {
+
+            String url = conceptpowerURL + conceptpowerSearchKeyword
+                    + URLEncoder.encode(searchString, StandardCharsets.UTF_8.toString()) + "&page=" + page;
+            URI uri = UriComponentsBuilder.fromUriString(url.toString()).build(true).toUri();
+
+            request = RequestEntity.get(uri).accept(MediaType.APPLICATION_JSON).build();
+        } catch (UnsupportedEncodingException e) {
+            throw new URISyntaxException(e.getMessage(), e.toString());
+
+        }
+        ResponseEntity<ConceptpowerResponse> response = null;
+        try {
+            response = restTemplate.exchange(request, ConceptpowerResponse.class);
+        } catch (RestClientException ex) {
+            throw new AuthorityServiceConnectionException(ex);
+        }
+
+        List<IAuthorityEntry> authorityEntries = new ArrayList<IAuthorityEntry>();
+        if (response.getStatusCode() == HttpStatus.OK) {
+            ConceptpowerResponse conceptEntries = response.getBody();
+            if (conceptEntries.getConceptEntries() != null) {
+                for (ConceptpowerEntry conceptEntry : conceptEntries.getConceptEntries()) {
+
+                    IAuthorityEntry authority = new AuthorityEntry();
+                    authority.setName(conceptEntry.getLemma());
+                    authority.setUri(conceptEntry.getEqual_to());
+                    authority.setDescription(conceptEntry.getDescription());
+                    authorityEntries.add(authority);
+                }
+            }
+        } else {
+            throw new AuthorityServiceConnectionException(response.getStatusCode().toString());
+        }
+        return authorityEntries;
+    }
+
+    @Override
+    public long totalRetrievedAuthorityData(String searchString)
+            throws URISyntaxException, AuthorityServiceConnectionException {
```

Comment:
by making this an extra method you on the one hand have a lot of duplicate code here, but also, you are making two requests where one would suffice since the call returns the total number of records, doesn't it? there shouldn't be an extra call, just for that.